### PR TITLE
feat: upgrade to masto.js v6

### PR DESCRIPTION
**Pull request by patak-dev**
Check out [the migration guide to see what's new in masto.js v6](https://github.com/neet/masto.js/releases/tag/v6.0.0)

The new rest API is quite nice (most of the changes in the PR are related to it). And the way streaming works also changed significantly. As far as I can tell, everything is working as expected (except for search, see https://github.com/neet/masto.js/issues/1018. Pushing the PR as a draft until the fixes for are available as a new release).

### CODE REVIEW

File: components/account/AccountHeader.vue [hunk 1]
```diff
@@ -53,7 +53,7 @@ function previewAvatar() {
 async function toggleNotifications() {
   relationship!.notifying = !relationship?.notifying
   try {
-    const newRel = await client.v1.accounts.follow(account.id, { notify: relationship?.notifying })
+    const newRel = await client.v1.accounts.$select(account.id).follow({ notify: relationship?.notifying })
```
Comment (Shinigami92) [inline components/account/AccountHeader.vue:56]: Why is there a `?` behind `relationship`, while on line 54 it is specifically given?
If `relationship` would be nullish on line 54, we would already run into a runtime error and never reach line 56

File: components/notification/NotificationCard.vue [hunk 2]
```diff
@@ -15,7 +15,6 @@ const { notification } = defineProps<{
           ps-3 pe-4 inset-is-0
           rounded-ie-be-3
           py-3 bg-base top-0
-          :lang="notification.status?.language ?? undefined"
```
Comment (Shinigami92) [inline components/notification/NotificationCard.vue:18]: Could you explain why this got removed? also below

**Review by Shinigami92 — COMMENTED** [review]
generally LGTM

**Comment by userquin** [comment]
We need to review types and logic in service-worker (@patak-dev I've sent you private DM in discord)

File: components/notification/NotificationCard.vue (hunk 2 above)
Comment (patak-dev) [reply components/notification/NotificationCard.vue:18]: A `follow` notification never has a `status`, same for the others were it was removed. The types in masto v6 may better reflect that and it was erroring out after the upgrade.

File: components/account/AccountHeader.vue (hunk 1 above)
Comment (patak-dev) [reply components/account/AccountHeader.vue:56]: Ya, line 54 also doesn't need it on the right side. We can do a PR to remove this later as it is unrelated to the upgrade.

**Comment by patak-dev** [comment]
> We need to review types and logic in service-worker (@patak-dev I've sent you private DM in discord)

The line you shared in Discord seems ok after the upgrade as it is a direct call. It doesn't use masto. If you could check anyways that things looks good on the service-worker/pwa side after this PR, appreciated!

**Comment by userquin** [comment]
@patak-dev it seems to be fine: 

imagen

**Review by userquin — APPROVED** [review]
sw logic seems to be fine

**Comment by patak-dev** [comment]
Search is fixed now. Tests are failing because of a type error on `AbortSignal.timeout`, but I think it may be an issue with Vitest? Node 18 should have that function

**Review by Shinigami92 — APPROVED** [review]

**Comment by patak-dev** [comment]
> Search is fixed now. Tests are failing because of a type error on `AbortSignal.timeout`, but I think it may be an issue with Vitest? Node 18 should have that function

Fixed this one by [polyfilling `AbortSignal.timeout` during test setup](https://github.com/elk-zone/elk/pull/2530/commits/e0e4ce1bd6aa0fbc7763e6233a2bc06041e19339).

I think we should be good to go now.

**Comment by patak-dev** [comment]
Let's merge this one so we try it out in https://main.elk.zone

**Comment by shuuji3** [comment]
Thanks for this upgrade. I can proceed #2530 with this new version. 👍🏻

**Pull request merged**